server/server.go

```diff
@@ -91,6 +91,7 @@ type Server struct {
 func NewServer(cfg *Config) (*Server, error) {
 	s, err := CreateServer(cfg)
 	if err != nil {
+		s.Close()
```

Comment:
No, `s` never be `nil`.

https://github.com/pingcap/pd/blob/c06a1ebbd81d1cca43bc1f76deac036ac2a4b135/server/server.go#L100